### PR TITLE
fix: support different lookup fields and custom_attributes for rETL events 

### PR DESCRIPTION
**Pull request by ItsSudip**
## What are the changes introduced in this PR?

We were not supporting random fields for lookups and for rETL events we are not sending custom_attributes properly. With this PR we are fixing these things.

## What is the related Linear task?

Resolves INT-2684

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3751/test-report.html

File: src/cdk/v2/destinations/intercom/procWorkflow.yaml [hunk 1]
```diff
@@ -77,8 +77,8 @@ steps:
     template: |
       const payload = .message.context.mappedToDestination ? $.outputs.rEtlPayload : $.outputs.identifyTransformationForLatestVersion;
       payload.name = $.getName(.message);
-      payload.custom_attributes = .message.context.traits || {};
-      payload.custom_attributes = $.filterCustomAttributes(payload, "user", .destination);
+      payload.custom_attributes = !.message.context.mappedToDestination ? .message.context.traits || {} : .message.traits.custom_attributes || {};
```
Comment (koladilip) [inline src/cdk/v2/destinations/intercom/procWorkflow.yaml:80]: payload.custom_attributes = (.message.context.mappedToDestination ? .message.traits.custom_attributes : .message.context.traits) || {};

Comment (koladilip) [reply src/cdk/v2/destinations/intercom/procWorkflow.yaml:80]: can be rewritten like this

File: src/cdk/v2/destinations/intercom/utils.js [hunk 2]
```diff
@@ -270,7 +276,10 @@ const filterCustomAttributes = (payload, type, destination) => {
  */
 const searchContact = async (message, destination, metadata) => {
   const lookupField = getLookUpField(message);
-  const lookupFieldValue = getFieldValueFromMessage(message, lookupField);
+  let lookupFieldValue = getFieldValueFromMessage(message, lookupField);
+  if (!lookupFieldValue) {
+    lookupFieldValue = get(message, `context.traits.${lookupField}`);
```
Comment (koladilip) [inline src/cdk/v2/destinations/intercom/utils.js:281]: lookupFieldValue = message?.context?.traits?.[lookupField]

**Review by koladilip — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Comment by ItsSudip** [comment]
@coderabbitai review

**Pull request merged**